samza-kv/src/main/scala/org/apache/samza/storage/kv/SerializedKeyValueStore.scala

```diff
@@ -59,23 +59,23 @@ class SerializedKeyValueStore[K, V](
     val valBytes = toBytesOrNull(value, msgSerde)
     store.put(keyBytes, valBytes)
     metrics.puts.inc
-    metrics.maxRecordSizeBytes.set(Math.max(metrics.maxRecordSizeBytes.getValue, valBytes.length))
+    while({val max = metrics.maxRecordSizeBytes.getValue; !metrics.maxRecordSizeBytes.compareAndSet(max, Math.max(max, valBytes.length))}){}
```

Comment:
`compareAndSet` probably doesn't have much overhead, but it might help to just check if the new `valBytes` is bigger than the current max (if not, then you don't even need to try the while loop). Most of the time, `valBytes` won't be bigger, so you won't have to set the value.

Reply:
good idea, will update this